x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/saml/SamlRealmTests.java

```diff
@@ -358,7 +358,9 @@ public void testAuthenticateWithRoleMapping() throws Exception {
             principalIsEmailAddress,
             populateUserMetadata,
             false,
-            authenticatingRealm
+            authenticatingRealm,
+            List.of("STRIKE Team: Delta$shield"),
+            "$"
```

Comment:
Nit: in order to preserve coverage for authenticating without delimiter splitting, I'd randomize here to pick between the two scenarios: no delimiter with a group list like `"avengers"`, `"shield"` and with delimiter with singleton group `"STRIKE Team: Delta$shield"`  

`testAuthenticateWithAuthorizingRealm` follows a separate flow that doesn't cover the same code path I don't think.